frontend/lib/posting_view/view/posting.view.dart

```diff
@@ -47,14 +57,66 @@ class _PostingViewState extends State<PostingView> {
               return itemUnavailable;
             }
 
-            final itemImage = item.imageUrl != null
-                ? Image.network(
-                    item.imageUrl!,
-                    width: double.infinity,
-                    height: 250,
-                    fit: BoxFit.cover,
-                  )
-                : Container();
+            Widget buildImageWidget(String url) {
+              return FutureBuilder<ImageProvider?>(
+                future: _fetchImage(url),
+                builder: (context, snapshot) {
+                  if (snapshot.connectionState == ConnectionState.waiting) {
+                    return const Center(child: CircularProgressIndicator());
+                  } else if (snapshot.hasError || !snapshot.hasData) {
+                    return const Center(
+                      child: Icon(
+                        Icons.broken_image,
+                        size: 50,
+                        color: Colors.grey,
+                      ),
+                    );
+                  } else {
+                    return Container(
+                      height: 250.0,
+                      decoration: BoxDecoration(
+                        borderRadius: BorderRadius.circular(4.0),
+                        image: DecorationImage(
+                          image: snapshot.data!,
+                          fit: BoxFit.cover,
+                        ),
+                      ),
+                    );
+                  }
+                },
+              );
+            }
+
+            Widget buildImageCarousel() {
+              if (item.pictures != null && item.pictures!.isNotEmpty) {
+                return SizedBox(
+                  height: 250.0,
+                  child: PageView.builder(
+                    itemCount: item.pictures!.length,
+                    itemBuilder: (context, index) {
+                      return buildImageWidget(item.pictures![index]);
```

Comment:
To add some tolerance, let's add a check here that returns the default image if for whatever reason pictures contains a null.